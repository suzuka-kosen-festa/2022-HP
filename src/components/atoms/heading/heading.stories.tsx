import type { ComponentStoryObj, ComponentMeta } from "@storybook/react";
import { Heading } from ".";

type T = typeof Heading;
type Story = ComponentStoryObj<T>;

const data = {
  sentence: "This is test",
};

export default {
  args: { children: data.sentence },
  argTypes: {
    children: { description: "Context of heading", control: { type: "text" } },
    color: {
      description: "First letter color",
      control: { type: "select", options: ["yellow", "green", "blue", "purple", "red"] },
    },
  },
  component: Heading,
} as ComponentMeta<T>;

export const Default: Story = {};
export const Green: Story = {
  args: {
    color: "green",
  },
};
export const Blue: Story = {
  args: {
    color: "blue",
  },
};
export const Purple: Story = {
  args: {
    color: "purple",
  },
};
export const Red: Story = {
  args: {
    color: "red",
  },
};
