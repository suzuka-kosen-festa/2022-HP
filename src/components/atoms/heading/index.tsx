import type { FC } from "react";
import tw, { css } from "twin.macro";
import type { TwStyle } from "twin.macro";
import type { HeadingProperties, ColorVariant } from "./type/model";

const HeadingBase = tw.h1`font-zen text-text text-5xl select-none`;

// eslint-disable-next-line consistent-return
const getColor = (color: ColorVariant): TwStyle => {
  switch (color) {
    case "yellow":
      return tw`text-primary-yellow`;
    case "green":
      return tw`text-primary-green`;
    case "blue":
      return tw`text-primary-blue`;
    case "purple":
      return tw`text-primary-purple`;
    case "red":
      return tw`text-primary-red`;
    // no default
  }
};

const Heading: FC<HeadingProperties> = ({ color = "yellow", children, ...rest }) => (
  <HeadingBase
    css={css`
      &::first-letter {
        ${getColor(color)}
      }
    `}
    {...rest}
  >
    {children}
  </HeadingBase>
);

export { Heading };
