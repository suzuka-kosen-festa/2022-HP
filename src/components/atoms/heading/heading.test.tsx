import { composeStories } from "@storybook/testing-react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import * as stories from "./heading.stories";

const { Default, Blue, Green, Red, Purple } = composeStories(stories);

const options = {
  name: "This is test",
};

describe("(components) atoms/heading", () => {
  test("to be atoms", () => {
    const { container } = render(<Default />);
    expect(container).toBeAtom();
  });
  test("to be [role=heading]", () => {
    const { getByRole } = render(<Default />);
    expect(getByRole("heading", options));
  });
  test("(default) take snap shot", () => {
    const { container } = render(<Default />);
    expect(container).toMatchSnapshot();
  });
  test("(blue) take snap shot", () => {
    const { container } = render(<Blue />)
    expect(container).toMatchSnapshot();
  })
  test("(green) take snap shot", () => {
    const { container } = render(<Green />)
    expect(container).toMatchSnapshot();
  })
  test("(red) take snap shot", () => {
    const { container } = render(<Red />)
    expect(container).toMatchSnapshot();
  })
  test("(purple) take snap shot", () => {
    const { container } = render(<Purple />)
    expect(container).toMatchSnapshot();
  })
});
