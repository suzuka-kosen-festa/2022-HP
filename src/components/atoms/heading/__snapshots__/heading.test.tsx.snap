// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(components) atoms/heading take snap shot 1`] = `
<div>
  <h1
    class="twin-f7kb4u-Heading ebaq0s80"
  >
    This is test
  </h1>
</div>
`;
